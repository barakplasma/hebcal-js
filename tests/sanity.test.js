const Hebcal = require("../src/hebcal");

const createNewHebcal = () => new Hebcal();

describe("Sanity Test", () => {
    let hebcal = createNewHebcal();
    beforeEach(() => {
        const hebcal = createNewHebcal();
    });
    describe("Initialize Hebcal instance", () => {
        test("that hebcal can be initialized without throwing an error", () => {
            expect(createNewHebcal).not.toThrow();
        });
        test("hebcal instance should be an instance of the base object ", () => {
            expect(hebcal).toBeInstanceOf(Hebcal);
        });
        test("should allow changing the city after initialization", () => {
            // Arrange
            const newYorkCalendar = createNewHebcal().setCity("New_York");
            // Action
            const jerusalemCalendar = hebcal.setCity("Jerusalem");
            // Assert
            expect(newYorkCalendar).not.toEqual(jerusalemCalendar);
        });
        test("should allow initialization using a Date object", () => {
            // Arrange
            const anHDate = new Hebcal.HDate(new Date(2014, 0, 1));
            // Assert
            expect(anHDate).toBeInstanceOf(Hebcal.HDate);
            expect(anHDate).toMatchSnapshot();
        });
        test("should initialize an HDate with a Hebrew date string", () => {
            // Arrange
            const specificDate = new Hebcal.HDate("1 Tishrei 5774");
            // Assert
            expect(specificDate).toMatchSnapshot();
        });
    });
    describe("Defaults", () => {
        test("should be initialized with this year as typeof number", () => {
            // Arrange
            const thisYear = hebcal.months[0].year;
            // Action
            const typeofYear = typeof thisYear;
            // Assert
            expect(typeofYear).toMatch("number");
            expect(thisYear).toBeGreaterThanOrEqual(5780);
        });
    });
    describe("Converting Dates", () => {
        test("should convert from gregorian year to hebrew year", () => {
            // Arrange
            const hebcalInstance = new Hebcal.GregYear(2017, 8);
            // Assert
            expect(hebcalInstance).toHaveProperty(["hebyears", 0], 5777);
        });
    });
    describe("Finding the date of a holiday", () => {
        test("should have a holiday property containing holidays for a given year, with a description for rosh hashana", () => {
            // Arrange
            const specificYear = new Hebcal(5777);
            // Assert
            expect(specificYear).toHaveProperty(
                ["holidays", "1 Tishrei 5777", 0, "desc", 0],
                "Rosh Hashana 1"
            );
        });
    });
});
