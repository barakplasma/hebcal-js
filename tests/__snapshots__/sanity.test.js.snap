// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanity Test Initialize Hebcal instance should allow initialization using a Date object 1`] = `
HDate {
  "day": 29,
  "il": false,
  "lat": 0,
  "long": 0,
  "month": 10,
  "year": 5774,
}
`;

exports[`Sanity Test Initialize Hebcal instance should initialize an HDate with a Hebrew date string 1`] = `
HDate {
  "day": 1,
  "il": false,
  "lat": 0,
  "long": 0,
  "month": 7,
  "year": 5774,
}
`;
